File: calc-refactor/typescript/tests/calculatorBuilder.ts
import { Calculator } from '../src/calculator.js';

export class CalculatorBuilder {
  private keys = '';

  pressKeys(keys: string): this {
    this.keys += keys;
    return this;
  }

  build(): Calculator {
    const calculator = new Calculator();
    for (const key of this.keys) calculator.press(key);
    return calculator;
  }
}

export function aCalculator(): CalculatorBuilder {
  return new CalculatorBuilder();
}

File: calc-refactor/python/tests/calculator_builder.py
from calc_refactor import Calculator


class CalculatorBuilder:
    def __init__(self) -> None:
        self._keys: str = ""

    def press_keys(self, keys: str) -> "CalculatorBuilder":
        self._keys += keys
        return self

    def build(self) -> Calculator:
        calculator = Calculator()
        for key in self._keys:
            calculator.press(key)
        return calculator


def a_calculator() -> CalculatorBuilder:
    return CalculatorBuilder()
